plugins/kubernetes-backend/src/cluster-locator/GkeClusterLocator.ts

```diff
@@ -55,8 +60,17 @@ export class GkeClusterLocator implements KubernetesClustersSupplier {
     );
   }
 
+  async getClusters(): Promise<ClusterDetails[]> {
```

Comment:
I'm not quite happy with this function.  I expect it works, but is there a more elegant way to cache/memoize this?  I did not find other examples in the codebase that do something similar.